statsmodels/multivariate/tests/test_multivariate_ols.py

```diff
@@ -105,8 +105,73 @@ def test_independent_variable_singular():
     mod = _MultivariateOLS.from_formula(
         'Histamine0 + Histamine1 + Histamine3 + Histamine5 ~ Drug * dup',
         data1, method='svd')
-    assert_raises_regex(ValueError, 'Covariance of x singular!', mod.fit)
+    assert_raises(ValueError, mod.fit)
     mod = _MultivariateOLS.from_formula(
         'Histamine0 + Histamine1 + Histamine3 + Histamine5 ~ Drug * dup',
         data1, method='pinv')
-    assert_raises_regex(ValueError, 'Covariance of x singular!', mod.fit)
+    assert_raises(ValueError,  mod.fit)
+
+
+def test_from_formula_vs_no_formula():
+    mod = _MultivariateOLS.from_formula(
+        'Histamine0 + Histamine1 + Histamine3 + Histamine5 ~ Drug * Depleted',
+        data)
+    r = mod.fit(method='svd')
+    r0 = r.mv_test()
+    endog, exog = patsy.dmatrices(
+        'Histamine0 + Histamine1 + Histamine3 + Histamine5 ~ Drug * Depleted',
+        data, return_type="dataframe")
```

Comment:
in general (not important here): we can reuse the endog and exog of a model (as long as we don't change them inplace)
e.g. from the previously defined model
`endog, exog = mod.endog, mod.exog`